Assets/Scripts/SS3D/Systems/Tile/PlacedObjects/PlacedTileObject.cs

```diff
@@ -0,0 +1,145 @@
+using FishNet;
+using FishNet.Object;
+using SS3D.Core;
+using SS3D.Logging;
+using SS3D.Systems.Tile.Connections;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace SS3D.Systems.Tile
+{
+    public class PlacedTileObject: NetworkBehaviour
+    {
+        /// <summary>
+        /// Creates a new PlacedTileObject from a TileObjectSO at a given position and direction. Uses NetworkServer.Spawn() if a server is running.
+        /// </summary>
+        /// <param name="worldPosition"></param>
+        /// <param name="dir"></param>
+        /// <param name="tileObjectSo"></param>
+        /// <returns></returns>
+        public static PlacedTileObject Create(Vector3 worldPosition, Vector2Int origin, Direction dir, TileObjectSo tileObjectSo)
+        {
+            GameObject placedGameObject = Instantiate(tileObjectSo.prefab);
+            placedGameObject.transform.SetPositionAndRotation(worldPosition, Quaternion.Euler(0, TileHelper.GetRotationAngle(dir), 0));
+
+            PlacedTileObject placedObject = placedGameObject.GetComponent<PlacedTileObject>();
+            if (placedObject == null)
+            {
+                placedObject = placedGameObject.AddComponent<PlacedTileObject>();
```

Comment:
In what case can this happen ? Bad set up in editor ? I'd rather avoid any AddComponent if this is the only case it can happen. If it is the case, put a PunPun Panic message instead and return.

Reply:
Same as with items, in most cases this component is added at runtime. Setting it up in the inspector would be double work since the prefab is configured in the corresponding SO.

Reply:
Same answer as before : Would it be possible to add it automatically but not at runtime ? Maybe using a custom inspector for a particular script ? If it's too tedious or you can't find anything simple, simply add a comment containing something along the lines of your answer to this conversation, as well as a statement that it'd be better if the component did not need to be added and we need to find a better solution.

Reply:
See above answer